mm2src/coins/rpc_command/get_estimated_fees.rs

```diff
@@ -0,0 +1,413 @@
+    fn status_code(&self) -> StatusCode {
+        match self {
+            FeeEstimatorError::CoinNotFoundOrSupported
+            | FeeEstimatorError::CoinNotConnected
+            | FeeEstimatorError::AlreadyStarted
+            | FeeEstimatorError::AlreadyStopping
+            | FeeEstimatorError::NotRunning
+            | FeeEstimatorError::CannotStartFromStopping => StatusCode::BAD_REQUEST,
+            FeeEstimatorError::Transport(_) | FeeEstimatorError::InternalError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<NumConversError> for FeeEstimatorError {
+    fn from(e: NumConversError) -> Self { FeeEstimatorError::InternalError(e.to_string()) }
+}
+
+impl From<String> for FeeEstimatorError {
+    fn from(e: String) -> Self { FeeEstimatorError::InternalError(e) }
+}
+
+/// Gas fee estimator loop context,
+/// runs the fee per gas estimation loop according to EIP-1559
+///
+/// This FeeEstimatorContext handles rpc requests which start and stop gas fee estimation loop and handles the loop itself.
+/// FeeEstimatorContext maintains a set of eth coins or tokens using the estimator.
+/// The loop estimation starts when any eth coin or token calls the start rpc and stops when the last using coin or token calls the stop rpc.
+/// FeeEstimatorContext keeps the latest estimated gas fees and returns them as response to a requesting rpc
+pub struct FeeEstimatorContext {
+    estimated_fees: AsyncMutex<FeePerGasEstimated>,
+    run_state: AsyncMutex<FeeEstimatorState>,
+    /// coins that connected to loop and can get fee estimates
+    using_coins: AsyncMutex<HashSet<String>>,
+    /// receiver of signals to stop estimator (if it is not used)
+    stop_listener: Arc<AsyncMutex<FeeEstimatorStopListener>>,
+    /// sender of signal to stop estimator
+    stop_handle: FeeEstimatorStopHandle,
+    /// mm2 shutdown listener
+    shutdown_listener: AsyncMutex<Pin<Box<dyn Future<Output = ()> + Send + Sync>>>,
+}
+
+impl FeeEstimatorContext {
+    fn new(ctx: MmArc) -> Result<Self, String> {
+        let shutdown_listener = try_s!(ctx.graceful_shutdown_registry.register_listener());
+        let (tx, rx) = mpsc::channel(MAX_CONCURRENT_STOP_REQUESTS);
+        Ok(Self {
+            estimated_fees: Default::default(),
+            run_state: AsyncMutex::new(FeeEstimatorState::default()),
+            using_coins: AsyncMutex::new(HashSet::new()),
+            stop_listener: Arc::new(AsyncMutex::new(rx)),
+            stop_handle: tx,
+            shutdown_listener: AsyncMutex::new(Box::pin(shutdown_listener)),
+        })
+    }
+
+    fn from_ctx(ctx: MmArc) -> Result<Arc<FeeEstimatorContext>, String> {
+        Ok(try_s!(from_ctx(&ctx.clone().fee_estimator_ctx, move || {
+            FeeEstimatorContext::new(ctx)
+        })))
+    }
+
+    /// Fee estimation update period in secs, basically equals to eth blocktime
+    fn get_refresh_interval() -> f64 { FEE_ESTIMATOR_REFRESH_INTERVAL as f64 }
```

Comment:
Why not removing `FEE_ESTIMATOR_REFRESH_INTERVAL` and using const function?

like:

```
    const fn get_refresh_interval() -> f64 { 15. }
```

Reply:
done